marine/marine.py

```diff
@@ -154,3 +172,31 @@ def _add_or_get_filter(
 
     def __del__(self):
         self._marine.destroy_marine()
+
+    @classmethod
+    def _expand_macros(
```

Comment:
Did you analyze the performance of the macros? I recall it slowing down Marine quite significantly.

Since the `macros` field is usually the same (or from a basically closed set), a caching mechanism can speed this up nicely :)

Reply:
It does slow the performance of marine.
More percisely, it slows from around 15,000 pps to 12,500 pps.

I tried implementing a caching mechanism but that did not help the performance. I think this is due to the fact that we can only cache the action performed by `_expand_macros`, while `_collapse_macros` is performed per result.

Reply:
Doesn't seem too bad for me (but sucks of course).
Remove the caching mechanism if it doesn't help.